modules/net-lb-proxy-int/variables.tf

```diff
@@ -0,0 +1,279 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "address" {
+  description = "Optional IP address used for the forwarding rule."
+  type        = string
+  default     = null
+}
+
+variable "backend_service_config" {
+  description = "Backend service level configuration."
+  type = object({
+    affinity_cookie_ttl_sec         = optional(number)
+    connection_draining_timeout_sec = optional(number)
+    health_checks                   = optional(list(string), ["default"])
+    log_sample_rate                 = optional(number)
+    port_name                       = optional(string)
+    project_id                      = optional(string)
+    session_affinity                = optional(string, "NONE")
+    timeout_sec                     = optional(number)
+    connection_tracking = optional(object({
+      idle_timeout_sec          = optional(number)
+      persist_conn_on_unhealthy = optional(string)
+      track_per_session         = optional(bool)
+    }))
+    failover_config = optional(object({
+      disable_conn_drain        = optional(bool)
+      drop_traffic_if_unhealthy = optional(bool)
+      ratio                     = optional(number)
+    }))
+  })
+  default  = {}
+  nullable = false
+  validation {
+    condition = (var.backend_service_config == null || contains(["NONE", "CLIENT_IP"],
+      var.backend_service_config.session_affinity
+    ))
+    error_message = "Invalid session affinity value."
+  }
+}
+
+variable "backends" {
```

Comment:
Can you use a single `backend_service_config` variable, which includes a `backends` attribute?

Reply:
You modeled this onto the ILB module but newer ones embed backends into the backend service config.

Reply:
Done